### PR TITLE
Ast-classes

### LINKED ISSUE #8: Write "temporary AST" classes
To make using the Tree-Sitter-API in the Interpreter easier we need another layer of abstraction.

This abstraction should consist of "temporary" AST-Objects (similar to the once that already exist). It should not be necessary to copy the complete syntax tree from Tree-Sitter to this abstraction. Instead the objects should be created "on-the-fly" in the Interpreter and use the Tree-Sitter nodes internally.

### PR DESCRIPTION
**Pull request by CFU9**
Closes #8

### CODE REVIEW

**Comment by tn4799** [comment]
Do you know that you don't have to close and reopen an PR? Just commit to your branch and it will be automatically added to the PR if it is not closed. 
And the title and the description could be edited as well here at github

**Review by Lythenas — CHANGES_REQUESTED** [review]
I think generally it looks pretty good.

Can you please run the format script (and you should probably install the commit hook to do that automatically).

It may be worth merging some of the nodes. E.g.

- `VariableDeclaration` and `LocalVariableDeclaration` can be merged with a bool saying if it is local or not.
- `VariableDeclaration` and `VariableDeclarator` can be merged which might make it easier to use.

**Comment by Lythenas** [comment]
`include/MiniLua/tree_sitter_ast.hpp` should be moved to `std/details/` because it does not need to be part of our public api. `src/core/tree_sitter_ast.cpp` should also be moved to `src/details/` and should probably be just named `ast.cpp`.

File: src/details/ast.cpp [hunk 1]
```diff
@@ -97,7 +98,8 @@ auto Program::body() -> Body { return Body(program.named_children()); }
 };*/
 // BinaryOperation
 BinaryOperation::BinaryOperation(ts::Node node) : bin_op(node) {
-    if (node.type_id() != ts::NODE_BINARY_OPERATION || node.child_count() != 3) {
+    assert(node.child_count() == 3);
```
Comment (Lythenas) [inline src/details/ast.cpp:101]: the asserts should be after the if because if we pass in the wrong type of node the exception is more appropriate.

**Review by Lythenas — CHANGES_REQUESTED** [review]

**Review by Lythenas — APPROVED** [review]

**Pull request merged**